File: android/src/main/java/com/twillio/src2/TwilioVideoPreview.java
/**
 * Component for Twilio Video local views.
 * <p>
 * Authors:
 * Jonathan Chang <slycoder@gmail.com>
 */

package com.twillio.src2;
import com.facebook.react.uimanager.ThemedReactContext;

public class TwilioVideoPreview extends RNVideoViewGroup {

    private static final String TAG = "TwilioVideoPreview";

    public TwilioVideoPreview(ThemedReactContext themedReactContext) {
        super(themedReactContext);
        CustomTwilioVideoView.registerThumbnailVideoView(this.getSurfaceViewRenderer());
    }

    public void applyZOrder(boolean applyZOrder) {
        this.getSurfaceViewRenderer().applyZOrder(applyZOrder);
    }
}

File: android/src/main/java/com/twilio/views/TwilioVideoPreview.kt
/**
 * Component for Twilio Video local views.
 *
 *
 * Authors:
 * Jonathan Chang <slycoder></slycoder>@gmail.com>
 */
package com.twilio.views

import com.facebook.react.uimanager.ThemedReactContext
import com.twilio.views.CustomTwilioVideoView.Companion.registerThumbnailVideoView

class TwilioVideoPreview(themedReactContext: ThemedReactContext?) :
    RNVideoViewGroup(themedReactContext) {
    init {
        registerThumbnailVideoView(surfaceViewRenderer)
    }

    fun applyZOrder(applyZOrder: Boolean) {
        surfaceViewRenderer!!.applyZOrder(applyZOrder)
    }

    companion object {
        private const val TAG = "TwilioVideoPreview"
    }
}
